src/equicordplugins/goodPerson/index.ts

```diff
@@ -17,48 +17,136 @@
 
-function replaceBadNouns(content) {
-
-    const regex = new RegExp("\\b(" + badNouns.join("|") + ")\\b", "gi");
-
-    return content.replace(regex, function (match) {
-        const randomIndex = Math.floor(Math.random() * badNounsReplacements.length);
-        return badNounsReplacements[randomIndex];
-    });
-}
-
-function replaceBadVerbs(content) {
-
-    const regex = new RegExp("\\b(" + badVerbs.join("|") + ")\\b", "gi");
-
-    return content.replace(regex, function (match) {
-        const randomIndex = Math.floor(Math.random() * badVerbsReplacements.length);
-        return badVerbsReplacements[randomIndex];
-    });
-}
-
 export default definePlugin({
     name: "GoodPerson",
-    description: "Makes you a good person",
-    authors: [Devs.nin0dev],
+    description: "Makes you (or others) a good person",
+    authors: [Devs.nin0dev, Devs.mantikafasi],
     dependencies: ["MessageEventsAPI"],
-
+    settings: definePluginSettings({
+        incoming: {
+            type: OptionType.BOOLEAN,
+            description: "Filter incoming messages",
+            default: true
+        },
+        blockSexual: {
+            type: OptionType.BOOLEAN,
+            description: "Block sexual words/hornyspeak",
+            default: true
+        },
+        blockBrainrot: {
+            type: OptionType.BOOLEAN,
+            description: "Block things commonly said by Gen Alpha children",
+            default: true
+        },
+        blockSlurs: {
+            type: OptionType.BOOLEAN,
+            description: "Block targeted slurs",
+            default: true
+        },
+        blockInsults: {
+            type: OptionType.BOOLEAN,
+            description: "Block more general insults",
+            default: true
+        },
+        blockOthers: {
+            type: OptionType.BOOLEAN,
+            description: "Block words mantikafasi personally dislikes",
+            default: true
+        }
+    }),
     async start() {
-        this.preSend = addPreSendListener((channelId, msg) => {
-            const newContent = replaceBadVerbs(replaceBadNouns(msg.content));
+        this.preSend = addPreSendListener((_channelId, msg) => {
+            const newContent = this.replaceBadVerbs(this.replaceBadNouns(msg.content));
             msg.content = newContent;
         });
     },
-
     stop() {
         removePreSendListener(this.preSend);
+    },
+    getEnabledBadNouns() {
+        const thingToReturn: string[] = [];
+        if (this.settings.store.blockBrainrot) thingToReturn.push(...badNounsBrainrot);
+        if (this.settings.store.blockFurryspeak) thingToReturn.push(...badNounsFurry);
```

Comment:
half assedly removed this banlist from the main plugin, but forgot to remove its setting